fix: put nested list popover header outside list items

DeepakSap14 opened the pull request:
## Related Issue
Closes https://github.com/SAP/fundamental-styles/issues/2691

## Description
Currently header for nested list popover is inside the list(**edit**: this header is list header, not popover header). But Ngx implementation shows that, it need to be outside the list(**edit**: popover header will be outside).
So created class "__popover-header" which will apply on popover header, outside list.
Also fixed border issue because of this change.

## Screenshots

### Before:
<img width="1032" alt="Screenshot 2021-08-26 at 12 18 12 PM" src="https://user-images.githubusercontent.com/57661487/130914844-a747730e-1abb-415d-aa1b-98153d5c2eb1.png">


### After:
<img width="1107" alt="Screenshot 2021-08-26 at 12 18 59 PM" src="https://user-images.githubusercontent.com/57661487/130914856-6c590028-035a-4bb8-ba4c-df1f765c0fac.png">

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- [NA] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

DeepakSap14 commented on stories/side-navigation/side-navigation.stories.js:
```diff
@@ -1267,10 +1267,10 @@ export const nestedListPopover = () => `<div class="fddocs-container" style="mar
             </ul>
         </div>
         <div class="fd-popover__popper" aria-hidden="false" id="popoverA11">
+            <div class="fd-nested-list fd-nested-list--popover fd-nested-list--compact">
```
@mikerodonnell89,
please have a look into this.

DeepakSap14 replied:
@platon-rov. now the structure is like below. Only popover header will be outside list.
`.fd-nested-list__popover-header
.fd-nested-list
    > .fd-nested-list__header
    > .fd-nested-list__body
        > .fd-nested-list__item
    > .fd-nested-list__footer`

InnaAtanasova reviewed (changes requested):
Please reconsider your implementation. In one case it's about `fd-nested-list__popover-header` and in the other it's about `fd-nested-list__group-header`. Two totally different things.

DeepakSap14 commented:
> Please reconsider your implementation. In one case it's about `fd-nested-list__popover-header` and in the other it's about `fd-nested-list__group-header`. Two totally different things.

@InnaAtanasova 
I have created a class "fd-nested-list__popover-header" for **popover header**, earlier it was not there.
I have kept **group header** as it is.
I have created examples in Styles here and Ngx [PR](https://github.com/SAP/fundamental-ngx/pull/6387).
please review once again.

InnaAtanasova commented on src/_nested-list.scss:
```diff
@@ -84,6 +84,25 @@ $button: #{$fd-namespace}-button;
   &--popover {
     border-bottom: none;
 
+    .#{$block}__popover-header {
```
The problem is that the popover header in ngx does not look like the group header:
<img width="198" alt="Screen Shot 2021-08-31 at 10 59 39 AM" src="https://user-images.githubusercontent.com/39598672/131526662-841e7dc2-dbf8-431a-b74a-941a66cd4538.png">

The first item in the popover is the popover header.
It's totally different from this:
<img width="256" alt="Screen Shot 2021-08-31 at 11 00 51 AM" src="https://user-images.githubusercontent.com/39598672/131526822-58110984-da4e-46da-ba91-3c1f1b931fb3.png">


Have you checked the design before implementing it?

DeepakSap14 commented on src/_nested-list.scss:
```diff
@@ -84,6 +84,25 @@ $button: #{$fd-namespace}-button;
     border-bottom: none;
 
+    .#{$block}__popover-header {
+      @include fd-nested-list-popover-border-top-radius();
```
#{$block}__popover-header is not needed.
Popover header design is created using fd-bar element. so fd-bar styles can be used here.

DeepakSap14 commented on src/_nested-list.scss:
```diff
@@ -85,21 +85,10 @@ $button: #{$fd-namespace}-button;
     border-bottom: none;
 
     .#{$block}__group-header {
-      @include fd-nested-list-popover-border-top-radius();
```
assuming, nested list popover will always have popover header. so border rounding is not needed on group header,

DeepakSap14 commented on src/_nested-list.scss:
```diff
@@ -85,21 +85,10 @@ $button: #{$fd-namespace}-button;
     }
 
     .#{$block}__item {
-      &:first-child {
```
assuming, nested list popover will always have popover header. so border rounding for first list item will not be needed.

InnaAtanasova reviewed (changes requested):
Please follow the fund-ngx structure. You can't use the Popover header here.

DeepakSap14 commented:
> Please follow the fund-ngx structure. You can't use the Popover header here.

@InnaAtanasova 
i did not find design for nested-list/nested-list-popover in wiki.
I will be helpful, if you can provide the link if it is there.

for now, i have created `fd-nested-list__popover-header` class, and used `fd-nested-list--text-only` & `fd-nested-list__title` for matching with fundamental-ngx.
let me know if this looks correct otherwise design spec needed for this.

InnaAtanasova reviewed (changes requested):
You can't have `fd-nested-list fd-nested-list--text-only fd-nested-list--popover` as the ` fd-nested-list--text-only` class is intended for the items inside the side nav (nested list) which have no icons. 
Also, your popover header is not in a compact mode as the rest of the elements. 
<img width="218" alt="Screen Shot 2021-09-03 at 10 40 52 AM" src="https://user-images.githubusercontent.com/39598672/132023824-e6c1b34e-b117-469e-b05c-7641cedae6ae.png">

DeepakSap14 commented:
> You can't have `fd-nested-list fd-nested-list--text-only fd-nested-list--popover` as the ` fd-nested-list--text-only` class is intended for the items inside the side nav (nested list) which have no icons (all items of the list should not have icons).
> Also, your popover header is not in a compact mode as the rest of the elements.
> <img alt="Screen Shot 2021-09-03 at 10 40 52 AM" width="218" src="https://user-images.githubusercontent.com/39598672/132023824-e6c1b34e-b117-469e-b05c-7641cedae6ae.png">

Thanks Inna.
removed `fd-nested-list--text-only` from popover-header.
introduced compact behaviour for popover-header.

InnaAtanasova commented on stories/side-navigation/side-navigation.stories.js:
```diff
@@ -1303,7 +1307,13 @@ export const nestedListPopover = () => `<div class="fddocs-container" style="mar
         </div>
         <div class="fd-popover__popper" aria-hidden="false" id="popoverA12">
-            <ul class="fd-nested-list fd-nested-list--popover fd-nested-list--compact">
+            <div class="fd-nested-list fd-nested-list--popover">
+                <div class="fd-nested-list__popover-header">
+                    Popover Header
+                </div>
```
Please follow the markup from fund-ngx.

DeepakSap14 replied:
Can you please give more info/describe more.
In last change, when i tried to follow the Ngx markup [here](https://github.com/SAP/fundamental-styles/pull/2692#pullrequestreview-746182875). there was concern that we can not use `fd-nested-list--text-only`.
I am not sure, what changes i need to make now.

DeepakSap14 replied:
@InnaAtanasova 
I am not sure about what markup i need to take from Ngx. what i have i already done is taken from Ngx markup.
It looks fine to me.
for comparing purpose i had created https://github.com/SAP/fundamental-ngx/pull/6387 PR, so comparision will be better. which is already linked in this PR.

Also as there is no design available and i am not clarified on exactly what i need to make change.
I am closing this PR.

DeepakSap14 commented:
Also as there is no design available and i am not clarified on exactly what i need to make change.
So, I am closing this PR.
